plugin/pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -57,3 +67,77 @@ func (pfactory *PredicateMetadataFactory) GetMetadata(pod *v1.Pod, nodeNameToInf
 	}
 	return predicateMetadata
 }
+
+func (pfactory *PredicateMetadataFactory) getMatchingAntiAffinityTerms(pod *v1.Pod, nodeInfoMap map[string]*schedulercache.NodeInfo) ([]matchingPodAntiAffinityTerm, error) {
+	allNodeNames := make([]string, 0, len(nodeInfoMap))
+	for name := range nodeInfoMap {
+		allNodeNames = append(allNodeNames, name)
+	}
+
+	var lock sync.Mutex
+	var result []matchingPodAntiAffinityTerm
+	var firstError error
+	appendResult := func(toAppend []matchingPodAntiAffinityTerm) {
+		lock.Lock()
+		defer lock.Unlock()
+		result = append(result, toAppend...)
+	}
+	catchError := func(err error) {
+		lock.Lock()
+		defer lock.Unlock()
+		if firstError == nil {
+			firstError = err
+		}
+	}
+
+	processNode := func(i int) {
+		nodeInfo := nodeInfoMap[allNodeNames[i]]
+		node := nodeInfo.Node()
+		if node == nil {
+			catchError(fmt.Errorf("node not found"))
+			return
+		}
+		var nodeResult []matchingPodAntiAffinityTerm
+		for _, existingPod := range nodeInfo.PodsWithAffinity() {
+			affinity := schedulercache.ReconcileAffinity(existingPod)
+			if affinity == nil {
+				continue
+			}
+			for _, term := range getPodAntiAffinityTerms(affinity.PodAntiAffinity) {
+				namespaceSelector, err := metav1.LabelSelectorAsSelector(term.NamespaceSelector)
+				if err != nil {
+					glog.V(10).Infof("Cannot parse namespaceSelector on term %v for pod %v. Details %v",
+						term, podName(pod), err)
+					return
+				}
+				namespaceList, err := pfactory.namespaceLister.List(namespaceSelector)
+				if err != nil {
+					glog.V(10).Infof("Cannot list namespace on term %v for pod %v. Details %v",
+						term, podName(pod), err)
+					return
+				}
+				namespaces := sets.String{}
+				for _, ns := range namespaceList {
+					namespaces.Insert(ns.Name)
+				}
+				if len(namespaces) == 0 {
+					namespaces = priorityutil.GetNamespacesFromPodAffinityTerm(pod, &term)
+				}
```

Comment:
Looks like the following pattern is repeated several times in this file. Could you please make a function out of it and call it instead of repeating the code multiple times?

+				namespaceSelector, err := metav1.LabelSelectorAsSelector(term.NamespaceSelector)
 +				if err != nil {
 +					glog.V(10).Infof("Cannot parse namespaceSelector on term %v for pod %v. Details %v",
 +						term, podName(pod), err)
 +					return
 +				}
 +				namespaceList, err := pfactory.namespaceLister.List(namespaceSelector)
 +				if err != nil {
 +					glog.V(10).Infof("Cannot list namespace on term %v for pod %v. Details %v",
 +						term, podName(pod), err)
 +					return
 +				}
 +				namespaces := sets.String{}
 +				for _, ns := range namespaceList {
 +					namespaces.Insert(ns.Name)
 +				}
 +				if len(namespaces) == 0 {
 +					namespaces = priorityutil.GetNamespacesFromPodAffinityTerm(pod, &term)
 +				}